--- a/emacs/uim-keymap.el
+++ b/emacs/uim-keymap.el
@@ -75,19 +75,22 @@
 ;; 
 (defun uim-unbind-mouse-event ()
 
-  (mapcar
-   '(lambda (w)
-      (mapcar 
-       '(lambda (x)
-	  (mapcar 
-	   '(lambda (y)
-	      (let ((event (vector (append w x (list y)))))
-	    (define-key uim-mode-map event nil)
-	    (define-key uim-preedit-map event nil)
-	    ))
-	   '(mouse-1 mouse-2 mouse-3 mouse-4 mouse-5)))
-       uim-mouse-modifiers))
-   uim-generic-modifiers)
+  (dolist (w uim-generic-modifiers)
+    (dolist (x uim-mouse-modifiers)
+      (dolist (y '(mouse-1 mouse-2 mouse-3 mouse-4 mouse-5))
+	(let ((event (vector (append w x (list y)))))
+	  (define-key uim-mode-map event nil)
+	  (define-key uim-preedit-map event nil)
+          )
+        )
+      )
+    (dolist (x '(wheel-down double-wheel-down triple-wheel-down wheel-up double-wheel-up triple-wheel-up))
+      (let ((event (vector (append w (list x)))))
+	(define-key uim-mode-map event nil)
+	(define-key uim-preedit-map event nil)
+        )
+      )
+    )
   
   (define-key uim-mode-map [vertical-scroll-bar] nil)
   (define-key uim-mode-map [mode-line] nil)
@@ -172,7 +172,9 @@
 	 (define-key uim-mode-map [t] 'uim-process-input)
 	 (define-key uim-preedit-map [t] 'uim-process-input)
 
-	 (define-key uim-mode-map (kbd "ESC") 'uim-process-input)
+	 ;; FIXME: precise version which do not this mapping
+	 (define-key uim-mode-map (kbd "ESC")
+		     (if (< emacs-major-version 27) 'uim-process-input))
 
          (define-key uim-mode-map (kbd "C-c") nil)
          (define-key uim-mode-map (kbd "C-x") nil)
--- a/emacs/uim-var.el
+++ b/emacs/uim-var.el
@@ -126,7 +126,7 @@
 
 ;; Type of Emacs
 (defconst uim-xemacs (featurep 'xemacs))
-(defconst uim-emacs (string-match "^GNU Emacs" (emacs-version)))
+(defconst uim-emacs (integerp (string-match "^GNU Emacs" (emacs-version))))
 
 ;; Supported languages and encodings
 ;; ("UIM-Language" "Emacs-Language" Emacs-encoding "UIM-Encoding")
--- a/emacs/uim-leim.el
+++ b/emacs/uim-leim.el
@@ -62,7 +62,7 @@
 
   (let (im)
     ;; register inactivation function
-    (setq inactivate-current-input-method-function 'uim-leim-inactivate)
+    (setq deactivate-current-input-method-function 'uim-leim-inactivate)
 
     ;; get plain IM engine name from LEIM style name
     ;;  ex. "Japanese-anthy-uim" => "anthy"
@@ -72,6 +72,7 @@
       ;; switch IM after uim activation
       (if (not (equal im uim-current-im-engine))
 	  (uim-change-im im))
+      (uim-leim-update-label)
 
       (setq uim-leim-active t))))
 
@@ -80,7 +81,7 @@
 (defun uim-leim-reset ()
   (when uim-leim-active
     (message "uim.el: uim-leim-reset")
-    (inactivate-input-method)))
+    (deactivate-input-method)))
 
 
 (defun uim-leim-make-im-name (im)
@@ -156,12 +157,12 @@
 	    (lambda ()
 	      (when uim-leim-active
 		(message "uim.el: LEIM inactivated"
-			 (inactivate-input-method)))))
+			 (deactivate-input-method)))))
 
   (add-hook 'uim-buffer-init-hook 
 	    (lambda ()
 	      (add-hook 'change-major-mode-hook
-			'inactivate-input-method nil t)))
+			'deactivate-input-method nil t)))
 
   ;; 
   ;;uim-leim-update-label 
