--- a/configure.ac
+++ b/configure.ac
@@ -4,12 +4,12 @@ m4_define([UIM_PATCHLEVEL_VERSION], [9])
 m4_define([UIM_VERSION],
     UIM_MAJOR_VERSION.UIM_MINOR_VERSION.UIM_PATCHLEVEL_VERSION)
 
-AC_PREREQ(2.60b)
-AC_INIT([uim], [UIM_VERSION], [uim-en@googlegroups.com])
+AC_PREREQ([2.72])
+AC_INIT([uim],[UIM_VERSION],[uim-en@googlegroups.com])
 AC_CONFIG_MACRO_DIR([m4.generated])
 AC_CONFIG_SRCDIR([uim/uim.c])
 AC_CONFIG_HEADERS([uim/config.h])
-AM_INIT_AUTOMAKE([1.10 foreign dist-bzip2])
+AM_INIT_AUTOMAKE([1.17 foreign dist-bzip2])
 
 AC_SUBST(UIM_VERSION_MAJOR, UIM_MAJOR_VERSION)
 AC_SUBST(UIM_VERSION_MINOR, UIM_MINOR_VERSION)
@@ -18,7 +18,7 @@ AC_SUBST(UIM_VERSION_PATCHLEVEL, UIM_PAT
 # Enable GNU extensions such as asprintf(3), BSD-originated functions,
 # POSIX features and more on glibc (and some other
 # implementations). See features.h. This macro should be placed here.
-AC_GNU_SOURCE
+AC_USE_SYSTEM_EXTENSIONS
 
 # Checks for programs.
 AC_PROG_CXX
@@ -28,14 +28,15 @@ AC_PROG_OBJC
 
 if test "${GCC}" = "yes"; then
   AC_MSG_CHECKING([for gcc is 4 or earlier])
-  AC_TRY_COMPILE([
+  AC_COMPILE_IFELSE(
+    [AC_LANG_SOURCE([[
 #if __GNUC__ >= 5
 #  error GCC 5 or later
 #endif
-                 ],
-                 [],
-                 [gcc_4_or_earlier=yes],
-                 [gcc_4_or_earlier=no])
+    ]])],
+    [gcc_4_or_earlier=yes],
+    [gcc_4_or_earlier=no]
+  )
   AC_MSG_RESULT($gcc_4_or_earlier)
 else
   gcc_4_or_earlier=no
@@ -46,7 +47,7 @@ AC_PROG_LN_S
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT
 AC_PATH_PROG(SH, sh)
 AC_PATH_PROG(RSVG, rsvg-convert)
 AC_PATH_PROGS(MD5, md5 md5sum)
@@ -104,7 +105,7 @@ AM_CONDITIONAL(M17NLIB, test "x$with_m17
 # *** Tests for Anthy ***
 # ***********************
 AC_ARG_WITH(anthy,
-  AC_HELP_STRING([--without-anthy],
+  AS_HELP_STRING([--without-anthy],
     [do not build with libanthy]),
   [],
   [with_anthy=yes])
@@ -135,7 +136,7 @@ AM_CONDITIONAL(ANTHY, test "x$with_anthy
 # *** Tests for Anthy (UTF-8) ***
 # *******************************
 AC_ARG_WITH(anthy-utf8,
-  AC_HELP_STRING([--with-anthy-utf8],
+  AS_HELP_STRING([--with-anthy-utf8],
     [build with libanthy (UTF-8)]),
   [],
   [with_anthy_utf8=no])
@@ -268,7 +269,7 @@ AM_CONDITIONAL(EXPAT, test "x$expat_foun
 # *** Tests for OpenSSL ***
 # *************************
 AC_ARG_ENABLE(openssl,
-  AC_HELP_STRING([--enable-openssl],
+  AS_HELP_STRING([--enable-openssl],
     [enable OpenSSL]),
   [],
   [enable_openssl=no])
@@ -330,26 +331,31 @@ if test x"$have_x" != xdisabled && test
     saved_LIBS=$LIBS
     CFLAGS="$CFLAGS $XFT_CFLAGS"
     LIBS="$LIBS $XFT_LIBS"
-    AC_TRY_LINK([
-#include <X11/Xft/Xft.h>], [ XftFontClose(0, 0); return 1; ],
-      [
+    AC_LINK_IFELSE(
+      [AC_LANG_PROGRAM(
+        [[ #include <X11/Xft/Xft.h> ]],
+        [[ XftFontClose(0, 0); return 1; ]]
+      )], [
         AC_DEFINE(WITH_XFT, 1, [font antialiasing support])
         AC_MSG_CHECKING([Xft UTF-8 support])
-        AC_TRY_LINK([
-#include <X11/Xft/Xft.h>
-          ], [
-XftDrawStringUtf8(0, 0, 0, 0, 0, 0, 0); return 0;
-          ],
+        AC_LINK_IFELSE(
+          [AC_LANG_PROGRAM(
+            [[ #include <X11/Xft/Xft.h> ]],
+            [[ XftDrawStringUtf8(0, 0, 0, 0, 0, 0, 0); return 0; ]]
+          )], [
             AC_DEFINE(HAVE_XFT_UTF8_STRING, 1, "Xft UTF8 support")
-            AC_MSG_RESULT(yes),
-            AC_MSG_RESULT(no))
-      ],
-      [
+            AC_MSG_RESULT(yes)
+          ], [
+            AC_MSG_RESULT(no)
+          ]
+        )
+      ], [
         AC_MSG_RESULT([***Could not link with Xft. Install Xft if you want support for it***])
         XFT=false
         XFT_CFLAGS=
         XFT_LIBS=
-      ])
+      ]
+    )
     CFLAGS=$saved_CFLAGS
     LIBS=$saved_LIBS
   fi
@@ -360,8 +366,6 @@ AC_XKB
 AM_CONDITIONAL(XIM, test x"$use_xim" = xyes)
 AM_CONDITIONAL(WITH_XFT, test x"$XFT" = xtrue)
 
-# Checks for header files
-AC_HEADER_STDC
 AC_HEADER_STDBOOL
 AC_CHECK_HEADERS([stdint.h inttypes.h sys/inttypes.h sys/types.h])
 AC_CHECK_HEADERS([fcntl.h locale.h stdlib.h unistd.h errno.h])
@@ -391,7 +395,6 @@ AC_TYPE_SSIZE_T
 AC_TYPE_OFF_T
 
 AC_TYPE_PID_T
-AC_TYPE_SIGNAL
 AC_CHECK_TYPES(sig_atomic_t, , ,
   [ #include <signal.h> ])
 AC_CHECK_TYPES(sig_t, , ,
@@ -576,19 +579,27 @@ AC_CHECK_FUNCS([waitpid])
 AC_CHECK_FUNCS([daemon])
 
 # IRIX has a const char return value for gai_strerror()
-AC_CHECK_FUNCS(gai_strerror,[
-        AC_DEFINE(HAVE_GAI_STRERROR)
-        AC_TRY_COMPILE([
+AC_CHECK_FUNCS(gai_strerror, [
+  AC_DEFINE(HAVE_GAI_STRERROR)
+  AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
 
-const char *gai_strerror(int);],[
+const char *gai_strerror(int);
+      ]], [[
 char *str;
 
-str = gai_strerror(0);],[
-                AC_DEFINE(HAVE_CONST_GAI_STRERROR_PROTO, 1,
-                [Define if gai_strerror() returns const char *])])])
+str = gai_strerror(0);
+      ]]
+    )], [
+      AC_DEFINE(HAVE_CONST_GAI_STRERROR_PROTO, 1,
+        [Define if gai_strerror() returns const char *])
+    ]
+  )
+])
 
 # Check for broken snprintf
 if test "x$ac_cv_func_snprintf" = xyes; then
@@ -649,15 +660,23 @@ fi
 # check that the fmt argument is const char * or just char *.
 # This is only useful for when BROKEN_SNPRINTF
 AC_MSG_CHECKING([whether snprintf can declare const char *fmt])
-AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#include <stdio.h>
-	   int snprintf(char *a, size_t b, const char *c, ...) { return 0; }
-	   int main(void) { snprintf(0, 0, 0); }
-    ]])],
-   [AC_MSG_RESULT(yes)
+AC_COMPILE_IFELSE(
+  [AC_LANG_PROGRAM(
+    [[
+#include <stdio.h>
+int snprintf(char *a, size_t b, const char *c, ...) { return 0; }
+    ]],
+    [[ snprintf(0, 0, 0); ]]
+  )],
+  [
+    AC_MSG_RESULT(yes)
     AC_DEFINE(SNPRINTF_CONST, [const],
-              [Define as const if snprintf() can declare const char *fmt])],
-   [AC_MSG_RESULT(no)
-    AC_DEFINE(SNPRINTF_CONST, [/* not const */])])
+              [Define as const if snprintf() can declare const char *fmt])
+  ], [
+    AC_MSG_RESULT(no)
+    AC_DEFINE(SNPRINTF_CONST, [/* not const */])
+  ]
+)
 
 AC_SEARCH_LIBS(dlfunc,dl,
   [
@@ -676,84 +695,94 @@ AC_CHECK_FUNC(inet_aton, ,
 AC_SUBST(NETLIBS)
 
 dnl socket related
-AC_CACHE_CHECK([for struct sockaddr_storage], ac_cv_have_struct_sockaddr_storage, [
-	AC_TRY_COMPILE(
-		[
+AC_CACHE_CHECK([for struct sockaddr_storage], ac_cv_have_struct_sockaddr_storage,
+  [AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
 #include <sys/types.h>
 #include <sys/socket.h>
-		],
-		[ struct sockaddr_storage s; ],
-		[ ac_cv_have_struct_sockaddr_storage=yes ],
-		[ ac_cv_have_struct_sockaddr_storage=no ]
-	)
-])
+      ]],
+      [[ struct sockaddr_storage s; ]]
+    )],
+    [ ac_cv_have_struct_sockaddr_storage=yes ],
+    [ ac_cv_have_struct_sockaddr_storage=no  ]
+  )]
+)
 if test "x$ac_cv_have_struct_sockaddr_storage" = xyes; then
 	AC_DEFINE(HAVE_STRUCT_SOCKADDR_STORAGE, 1,
 		[define if you have struct sockaddr_storage data type])
 fi
 
 dnl cred related
 AC_CHECK_TYPES([struct sockcred], [], [],[[#include <sys/socket.h>]])
-AC_CACHE_CHECK([for struct cmsgcred], ac_cv_have_struct_cmsgcred, [
-	AC_TRY_COMPILE(
-		[
+AC_CACHE_CHECK([for struct cmsgcred], ac_cv_have_struct_cmsgcred,
+  [AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
 #include <sys/types.h>
 #include <sys/socket.h>
-		],
-		[ struct cmsgcred s; ],
-		[ ac_cv_have_struct_cmsgcred=yes ],
-		[ ac_cv_have_struct_cmsgcred=no ]
-	)
-])
+      ]],
+      [[ struct cmsgcred s; ]]
+    )],
+    [ ac_cv_have_struct_cmsgcred=yes ],
+    [ ac_cv_have_struct_cmsgcred=no  ]
+  )]
+)
 if test "x$ac_cv_have_struct_cmsgcred" = xyes; then
 	AC_DEFINE(HAVE_STRUCT_CMSGCRED, 1,
 		[define if you have struct cmsgcred data type])
 fi
 
-AC_CACHE_CHECK([for struct sockaddr_in6], ac_cv_have_struct_sockaddr_in6, [
-	AC_TRY_COMPILE(
-		[
+AC_CACHE_CHECK([for struct sockaddr_in6], ac_cv_have_struct_sockaddr_in6,
+  [AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
 #include <sys/types.h>
 #include <netinet/in.h>
-		],
-		[ struct sockaddr_in6 s; s.sin6_family = 0; ],
-		[ ac_cv_have_struct_sockaddr_in6=yes ],
-		[ ac_cv_have_struct_sockaddr_in6=no ]
-	)
-])
+      ]],
+      [[ struct sockaddr_in6 s; s.sin6_family = 0; ]]
+    )],
+    [ ac_cv_have_struct_sockaddr_in6=yes ],
+    [ ac_cv_have_struct_sockaddr_in6=no  ]
+  )]
+)
 if test "x$ac_cv_have_struct_sockaddr_in6" = xyes; then
 	AC_DEFINE(HAVE_STRUCT_SOCKADDR_IN6, 1,
 		[define if you have struct sockaddr_in6 data type])
 fi
 
-AC_CACHE_CHECK([for struct in6_addr], ac_cv_have_struct_in6_addr, [
-	AC_TRY_COMPILE(
-		[
+AC_CACHE_CHECK([for struct in6_addr], ac_cv_have_struct_in6_addr,
+  [AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
 #include <sys/types.h>
 #include <netinet/in.h>
-		],
-		[ struct in6_addr s; s.s6_addr[0] = 0; ],
-		[ ac_cv_have_struct_in6_addr=yes ],
-		[ ac_cv_have_struct_in6_addr=no ]
-	)
-])
+      ]],
+      [[ struct in6_addr s; s.s6_addr[0] = 0; ]]
+    )],
+    [ ac_cv_have_struct_in6_addr=yes ],
+    [ ac_cv_have_struct_in6_addr=no  ]
+  )]
+)
 if test "x$ac_cv_have_struct_in6_addr" = xyes; then
 	AC_DEFINE(HAVE_STRUCT_IN6_ADDR, 1,
 		[define if you have struct in6_addr data type])
 fi
 
-AC_CACHE_CHECK([for struct addrinfo], ac_cv_have_struct_addrinfo, [
-	AC_TRY_COMPILE(
-		[
+AC_CACHE_CHECK([for struct addrinfo], ac_cv_have_struct_addrinfo,
+  [AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
-		],
-		[ struct addrinfo s; s.ai_flags = AI_PASSIVE; ],
-		[ ac_cv_have_struct_addrinfo=yes ],
-		[ ac_cv_have_struct_addrinfo=no ]
-	)
-])
+      ]],
+      [[ struct addrinfo s; s.ai_flags = AI_PASSIVE; ]]
+    )],
+    [ ac_cv_have_struct_addrinfo=yes ],
+    [ ac_cv_have_struct_addrinfo=no  ]
+  )]
+)
 if test "x$ac_cv_have_struct_addrinfo" = xyes; then
 	AC_DEFINE(HAVE_STRUCT_ADDRINFO, 1,
 		[define if you have struct addrinfo data type])
@@ -804,7 +833,7 @@ dnl ******************************
 dnl *** Check for GTK+ Library ***
 dnl ******************************
 AC_ARG_WITH(gtk2,
-  AC_HELP_STRING([--without-gtk2],
+  AS_HELP_STRING([--without-gtk2],
     [do not build against GTK+2]),
   [],
   [with_gtk2=yes])
@@ -828,7 +857,7 @@ if test "x$with_gtk2" = xyes; then
 fi
 
 AC_ARG_ENABLE(gnome-applet,
-  AC_HELP_STRING([--disable-gnome-applet],
+  AS_HELP_STRING([--disable-gnome-applet],
     [disable uim applet for GNOME2 panel]),
   [],
   [enable_gnome_applet=yes])
@@ -851,7 +880,7 @@ dnl *******************************
 dnl *** Check for GTK+3 Library ***
 dnl *******************************
 AC_ARG_WITH(gtk3,
-  AC_HELP_STRING([--without-gtk3],
+  AS_HELP_STRING([--without-gtk3],
     [do not build GTK+3 immodule and helper applications]),
   [],
   [with_gtk3=yes])
@@ -871,7 +900,7 @@ if test "x$with_gtk3" = xyes; then
 fi
 
 AC_ARG_ENABLE(gnome3-applet,
-  AC_HELP_STRING([--disable-gnome3-applet],
+  AS_HELP_STRING([--disable-gnome3-applet],
     [disable uim applet for GNOME3 panel]),
   [],
   [enable_gnome3_applet=yes])
@@ -892,7 +921,7 @@ if test "x$enable_gnome3_applet" = xyes;
 fi
 
 AC_ARG_ENABLE(debug,
-  AC_HELP_STRING([--enable-debug],
+  AS_HELP_STRING([--enable-debug],
     [enable debugging]),
   [],
   [enable_debug=no])
@@ -903,7 +932,7 @@ dnl ****************************
 dnl *** Check for Qt Library ***
 dnl ****************************
 AC_ARG_WITH(qt,
-  AC_HELP_STRING([--with-qt],
+  AS_HELP_STRING([--with-qt],
     [build Qt3 helper applications]),
   [],
   [with_qt=no])
@@ -1011,7 +1040,7 @@ if test "x$with_qt" = xyes; then
   QT_LDFLAGS=-L${QTLIBDIR}
   LDFLAGS="$LDFLAGS $QT_LDFLAGS"
   AC_LANG_SAVE
-  AC_LANG_CPLUSPLUS
+  AC_LANG([C++])
   AC_CHECK_LIB(qt, main, QT_LIB=-lqt,
       AC_CHECK_LIB(qt-mt, main, QT_LIB=-lqt-mt,
           AC_MSG_ERROR([Cannot find QT libraries.])))
@@ -1030,12 +1059,19 @@ if test "x$with_qt" = xyes; then
   LIBS="$LIBS $QT_LIBS"
 
   AC_MSG_CHECKING(Qt - version >= $QT_VERSION)
-  AC_TRY_COMPILE([#include <qglobal.h>],
-  [
-      #if (QT_VERSION < $QT_VERSION_NUM)
-          #error  "QT_VERSION too old"
-      #endif
-  ],result=yes,result=no)
+  AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[
+#include <qglobal.h>
+      ]], [[
+#if (QT_VERSION < $QT_VERSION_NUM)
+    #error  "QT_VERSION too old"
+#endif
+      ]]
+    )],
+    [result=yes],
+    [result=no]
+  )
 
   AC_MSG_RESULT("$result")
   if test "x$result" = xno; then
@@ -1070,7 +1106,7 @@ fi
 AM_CONDITIONAL(QT, test "x$with_qt" = xyes)
 
 AC_ARG_WITH(qt-immodule,
-  AC_HELP_STRING([--with-qt-immodule],
+  AS_HELP_STRING([--with-qt-immodule],
     [build Qt3-immodule extension]),
   [],
   [with_qt_immodule=no])
@@ -1093,7 +1129,7 @@ dnl *****************************
 dnl *** Check for Qt4 Library ***
 dnl *****************************
 AC_ARG_WITH(qt4,
-  AC_HELP_STRING([--with-qt4],
+  AS_HELP_STRING([--with-qt4],
     [build Qt4 helper applications]),
   [],
   [with_qt4=no])
@@ -1101,7 +1137,7 @@ AC_ARG_WITH(qt4,
 AM_CONDITIONAL(QT4, test "x$with_qt4" = xyes)
 
 AC_ARG_WITH(qt4-immodule,
-  AC_HELP_STRING([--with-qt4-immodule],
+  AS_HELP_STRING([--with-qt4-immodule],
     [build Qt4-immodule extension]),
   [],
   [with_qt4_immodule=no])
@@ -1118,7 +1154,7 @@ dnl ************************************
 dnl *** test for Qt3Support module in Qt4 ***
 dnl *****************************************
 AC_ARG_ENABLE(qt4-qt3support,
-  AC_HELP_STRING([--disable-qt4-qt3support],
+  AS_HELP_STRING([--disable-qt4-qt3support],
     [disable support for Qt3Support module in Qt4]),
   [],
   [enable_qt4_qt3support=yes])
@@ -1140,7 +1176,7 @@ dnl *****************************
 dnl *** Check for Qt5 Library ***
 dnl *****************************
 AC_ARG_WITH(qt5,
-  AC_HELP_STRING([--with-qt5],
+  AS_HELP_STRING([--with-qt5],
     [build Qt5 helper applications]),
   [],
   [with_qt5=no])
@@ -1148,7 +1184,7 @@ AC_ARG_WITH(qt5,
 AM_CONDITIONAL(QT5, test "x$with_qt5" = xyes)
 
 AC_ARG_WITH(qt5-immodule,
-  AC_HELP_STRING([--with-qt5-immodule],
+  AS_HELP_STRING([--with-qt5-immodule],
     [build Qt5-immodule extension]),
   [],
   [with_qt5_immodule=no])
@@ -1173,7 +1209,7 @@
 dnl *** Check for Qt6 Library ***
 dnl *****************************
 AC_ARG_WITH(qt6,
-  AC_HELP_STRING([--with-qt6],
+  AS_HELP_STRING([--with-qt6],
     [build Qt6 helper applications]),
   [],
   [with_qt6=no])
@@ -1176,7 +1212,7 @@ AC_ARG_WITH(qt6,
 AM_CONDITIONAL(QT6, test "x$with_qt6" = xyes)
 
 AC_ARG_WITH(qt6-immodule,
-  AC_HELP_STRING([--with-qt6-immodule],
+  AS_HELP_STRING([--with-qt6-immodule],
     [build Qt6-immodule extension]),
   [],
   [with_qt6_immodule=no])
@@ -1162,7 +1198,7 @@ fi
 AM_CONDITIONAL(QT6_IMMODULE, test "x$with_qt6_immodule" = xyes)
 
 AC_ARG_ENABLE(default-toolkit,
-  AC_HELP_STRING([--enable-default-toolkit],
+  AS_HELP_STRING([--enable-default-toolkit],
     [determine default toolkit (gtk, gtk3, qt, qt4, qt5 or qt6) @<:@default=auto@:>@]),
   [],
   [enable_default_toolkit=auto])
@@ -1195,7 +1231,7 @@ AM_CONDITIONAL(DEFAULT_TOOLKIT_QT5, test
 AC_SUBST(default_toolkit)
 
 AC_ARG_ENABLE(fep,
-  AC_HELP_STRING([--disable-fep],
+  AS_HELP_STRING([--disable-fep],
     [disable uim-fep]),
   [],
   [enable_fep=yes])
@@ -1216,7 +1252,7 @@ fi
 AM_CONDITIONAL(UIM_FEP, test "x$enable_fep" = xyes)
 
 AC_ARG_ENABLE(emacs,
-  AC_HELP_STRING([--disable-emacs],
+  AS_HELP_STRING([--disable-emacs],
     [disable uim.el]),
   [],
   [enable_emacs=yes])
@@ -1229,7 +1265,7 @@ fi
 AM_CONDITIONAL(UIM_EL, test "x$enable_emacs" = xyes)
 
 AC_ARG_ENABLE(dict,
-  AC_HELP_STRING([--enable-dict],
+  AS_HELP_STRING([--enable-dict],
     [build dictionary utility for uim (Anthy and Canna)]),
   [],
   [enable_dict=no])
@@ -1324,7 +1360,7 @@ AM_CONDITIONAL(OSX_DCS, test "x$with_osx
 # This code was based on openssh-4.1p1
 
 AC_ARG_WITH(libedit,
-  AC_HELP_STRING([--with-libedit[=DIR]],
+  AS_HELP_STRING([--with-libedit[=DIR]],
     [enable libedit support @<:@default=yes@:>@]),
   [],
   [with_libedit=yes])
@@ -1379,7 +1415,7 @@ dnl **********************************
 dnl *** test for KDE3 applet.. XXX ***
 dnl **********************************
 AC_ARG_ENABLE(kde-applet,
-  AC_HELP_STRING([--disable-kde-applet],
+  AS_HELP_STRING([--disable-kde-applet],
     [do not build uim applet for KDE3 panel]),
   [],
   [enable_kde_applet=yes])
@@ -1424,7 +1460,7 @@ dnl ****************************
 dnl *** test for KDE4 applet ***
 dnl ****************************
 AC_ARG_ENABLE(kde4-applet,
-  AC_HELP_STRING([--disable-kde4-applet],
+  AS_HELP_STRING([--disable-kde4-applet],
     [do not build uim KDE4 Plasma widget]),
   [],
   [enable_kde4_applet=yes])
@@ -1468,7 +1504,7 @@ AC_ARG_ENABLE(qt5-applet,
 AM_CONDITIONAL(QT5_APPLET, test "x$enable_qt5_apple" = xyes)
 
 AC_ARG_ENABLE(pref,
-  AC_HELP_STRING([--disable-pref],
+  AS_HELP_STRING([--disable-pref],
     [do not build graphical utility to edit user settings]),
   [],
   [enable_pref=yes])
@@ -1498,7 +1534,7 @@ fi
 AM_CONDITIONAL(PREF, test x"$enable_pref" = xyes)
 
 AC_ARG_ENABLE(anthy-static,
-  AC_HELP_STRING([--enable-anthy-static],
+  AS_HELP_STRING([--enable-anthy-static],
     [enable static Anthy]),
   [],
   [enable_anthy_static=no])
@@ -1506,7 +1542,7 @@ AC_ARG_ENABLE(anthy-static,
 AM_CONDITIONAL(ENABLE_ANTHY_STATIC, test "x$enable_anthy_static" = xyes)
 
 AC_ARG_ENABLE(anthy-utf8-static,
-  AC_HELP_STRING([--enable-anthy-utf8-static],
+  AS_HELP_STRING([--enable-anthy-utf8-static],
     [enable static Anthy-UTF8]),
   [],
   [enable_anthy_utf8_static=no])
@@ -1670,7 +1706,7 @@ else
 fi
 
 AC_ARG_ENABLE(warnings-into-error,
-  AC_HELP_STRING([--enable-warnings-into-error],
+  AS_HELP_STRING([--enable-warnings-into-error],
     [treat compiler warnings as error @<:@default=no if !debug@:>@]),
     [
         case $enable_warnings_into_error in
